druid/src/core.rs

```diff
@@ -1082,6 +1088,17 @@ impl<T: Data, W: Widget<T>> WidgetPod<T, W> {
             // Update focus-chain of our parent
             LifeCycle::BuildFocusChain => {
                 self.state.update_focus_chain = false;
+
+                // had_focus is the old focus value. state.has_focus was repaced with ctx.is_focused().
+                // Therefore is had_focus is true but state.has_focus is false the widgets, which is
+                // currently focused is not part of the functional tree anymore
+                // (Lifecycle::BuildFocusChain.should_propagate_to_hidden() is false!) and should
+                // resign the focus.
+                if had_focus && !self.state.has_focus {
```

Comment:
I'm confused; we haven't done `merge_up` yet, so how would `has_focus` have changed?

Reply:
The `merge_up` of our children happened already in line 1058 / 1062. If someone hides a focused widget and then calls `children_changed()`, we clear `has_focus` in `BuildFocusChain` if has_clear is not set after the calls in 1058 and 1062 we know, that the focused widget is not reachable